mlrun/model_monitoring/stream_processing.py

```diff
@@ -783,6 +784,10 @@ def do(self, event: dict):
             if not feature_names and self._infer_columns_from_data:
                 feature_names = self._infer_feature_names_from_data(event)
 
+            is_router_endpoint = endpoint_record.get(
```

Comment:
I suggest moving it inside the if clause (if not feature_names: ...) since it is only relevant to that specific case.
Additionally, on L852 we call `endpoint_record.get(EventFieldType.ENDPOINT_TYPE)` again. It would be nice to merge these calls to avoid redundancy, though this is more of a nice-to-have improvement.

Reply:
But it's also needed inside the `if not label_columns:` block. Anyway, it doesn't matter following refactor based on your second point. Please check again.

Reply:
Notebook completed successfully.